1575 spice   update spin and repoint table to have utc columns

tmplummer opened the pull request:
# Change Summary

## Overview
Updates to the spin and repoint table definitions:
- UTC columns added
- "sclk" added to columns with spacecraft clock times

## Updated Files
- imap_processing/spice/repoint.py
- imap_processing/spice/spin.py
- imap_processing/tests/conftest.py
- imap_processing/tests/spice/test_data/fake_repoint_data.csv
- imap_processing/tests/spice/test_data/fake_spin_data.csv
- imap_processing/tests/spice/test_repoint.py
- imap_processing/tests/spice/test_spin.py

Closes: #1575

tmplummer committed:
Update repoint table to match new definition

tmplummer committed:
Update spin table to match new definition

greglucas commented on imap_processing/spice/repoint.py:
```diff
@@ -29,14 +29,18 @@ def get_repoint_data() -> pd.DataFrame:
-            - `repoint_start_sec`: Starting MET seconds of repoint maneuver.
-            - `repoint_start_subsec`: Starting MET milliseconds of repoint maneuver.
+            - `repoint_start_sec_sclk`: Starting MET seconds of repoint maneuver.
+            - `repoint_start_subsec_sclk`: Starting MET milliseconds of repoint
```
I think these are microseconds now and will be 6-digits long. At least that is what Ian put in his latest e-mail back to us in the spin file. Will need to update this and divide by 1e6 if you agree.

tmplummer replied:
Oh, good catch. I didn't read those emails carefully enough.

greglucas commented on imap_processing/tests/conftest.py:
```diff
@@ -523,18 +524,25 @@ def make_data(start_met: float, end_met: Optional[float] = None) -> pd.DataFrame
         spin_start_subsec = int((start_met - spin_start_sec[0]) * 1000)
 
+        # Calculate UTC times without spice (accepting ~5 second inaccuracy)
+        spin_start_dt64 = TTJ2000_EPOCH + (spin_start_met * 1e9).astype(
```
Can we load up the `utc` value here as a dt64? Then I think we would get the correct accuracy because they are using SPICE already.

edit: I see, we are in the testing directory, so these aren't the files we were given they are interal to us. Ignore this comment, just leaving it to note my lack of reading context properly!

tech3371 reviewed (commented):
Hi Tim,
I made some change on the Galaxy page based on what we saw in actual data. Can you update this PR to match some of those names? What Ian had the data was different from what we discussed in the email. I updated our page to match what's in the data. [page link here](https://lasp.colorado.edu/galaxy/spaces/IMAP/pages/173715570/Spin+and+Repoint+MOC+Data+product)

tmplummer committed:
Convert repoint subseconds field to microseconds

tmplummer committed:
Improve documentation

tmplummer commented:
> Hi Tim, I made some change on the Galaxy page based on what we saw in actual data. Can you update this PR to match some of those names? What Ian had the data was different from what we discussed in the email. I updated our page to match what's in the data. [page link here](https://lasp.colorado.edu/galaxy/spaces/IMAP/pages/173715570/Spin+and+Repoint+MOC+Data+product)

@tech3371, I think that I already did this. I downloaded the a couple of the spin and repoint csv files that were delivered and checked the column names against what I had in the fake csv files and updated any discrepancies. Other than the subseconds being microseconds issue that @greglucas pointed out, was there a specific discrepancy that you saw?

tech3371 commented on imap_processing/spice/repoint.py:
```diff
@@ -26,18 +26,22 @@ def get_repoint_data() -> pd.DataFrame:
+            * `repoint_end_subsec_sclk`: Ending MET microseconds of repoint maneuver.
+            * `repoint_end_met`: Floating point MET of repoint maneuver end time.
+              Derived from `repoint_end_sec_sclk` and `repoint_end_subsec_sclk`.
+            * `repoint_end_time_utc`: UTC time of repoint maneuver end time.
```
```suggestion
            * `repoint_end_utc`: UTC time of repoint maneuver end time.
```

tech3371 commented on imap_processing/spice/repoint.py:
```diff
@@ -26,18 +26,22 @@ def get_repoint_data() -> pd.DataFrame:
+              maneuver.
+            * `repoint_start_met`: Floating point MET of repoint maneuver start time.
+              Derived from `repoint_start_sec_sclk` and `repoint_start_subsec_sclk`.
+            * `repoint_start_time_utc`: UTC time of repoint maneuver start time.
```
Looks like we just need update in the doc string.
```suggestion
            * `repoint_start_utc`: UTC time of repoint maneuver start time.
```

tmplummer committed:
Fix broken ultra tests
Rename spin_start_time to spin_start_met for consistency
Fix broken hi test

tmplummer committed:
Search and replace repoint_start_time_utc and repoint_end_time_utc

tmplummer commented:
@laspsandoval, The definition changes to the spin and repoint tables required a few Ultra updates. Please review these when you get a chance.

tmplummer committed:
Fix more hi tests

laspsandoval commented on imap_processing/tests/ultra/unit/test_ultra_l1b_extended.py:
```diff
@@ -444,10 +444,12 @@ def test_get_eventtimes(test_fixture, use_fake_spin_data_for_time):
 
     spin_start_min = (
-        expected_min_df["spin_start_sec"] + expected_min_df["spin_start_subsec"] / 1000
+        expected_min_df["spin_start_sec_sclk"]
```
Is it being in microseconds something new?

tmplummer replied:
Yes, that was one of the many changes made last week.

laspsandoval reviewed (approved):
Looks good on the Ultra side of things.
